### PR TITLE
feat(Teleport): add dash teleport

### PR DESCRIPTION
**Pull request by mbbmbbmm**
Dashing Teleport inspired by the implementation of realities.io.
When dashing through objects their renderers can be turned off.
An example scene has been added to show it in action.

### CODE REVIEW

**Comment by thestonefox** [comment]
Looks good. I'll have a proper play when i'm back from vacation (on Saturday) so hopefully it will be a quick merge :)

File: Assets/VRTK/Scripts/VRTK_DashTeleport.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+
+        private IEnumerator lerpToPosition(Vector3 targetPosition, Transform target)
+        {
+            base.enableTeleport = false;
+            bool gameObjectInTheWay = false;
+
+            // Find the objects we will be dashing through and broadcast them via events
+            Vector3 eyeCameraPosition = eyeCamera.transform.position;
+            Vector3 eyeCameraPositionOnGround = new Vector3(eyeCameraPosition.x, this.transform.position.y, eyeCameraPosition.z);
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_DashTeleport.cs]: don't need to write `this.transform` should just be `transform`

File: Assets/VRTK/Scripts/VRTK_DashTeleport.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                lerpTime = (1 / minSpeedMps) * maxDistance; // clamped to speed for small dashes
+            }
+
+            Vector3 startPosition = new Vector3(transform.position.x, transform.position.y, transform.position.z);
+            float elapsedTime = 0;
+            float t = 0;
+
+            while (t < 1)
+            {
+                this.transform.position = Vector3.Lerp(startPosition, targetPosition, t);
```
Comment (thestonefox) [inline Assets/VRTK/Scripts/VRTK_DashTeleport.cs]: don't need to write `this.transform` should just be `transform`

**Comment by mbbmbbmm** [comment]
Yes I had used it intentionally for better understanding but looking at it again I'm not sure why I thought that it was helpful. It's pretty clear without it. Anyway I took it out now.

**Comment by thestonefox** [comment]
I used to use the `this` keyword as well (you'll see in earlier versions of the toolkit code it's there) but I was told not to put it in, so I took it out :)

**Comment by mbbmbbmm** [comment]
Oh I see. :) It's interesting, I hadn't really thought about how questions like this need to be resolved with an open source project.

**Comment by thestonefox** [comment]
It's pretty much ready to go, the one thing i'd say is in the example scene perhaps use the Bezier Pointer rather than the simple pointer (or at least on one of the controllers) what do you think?

**Comment by mbbmbbmm** [comment]
Good idea, I added the Bezier to one of the controllers!

**Comment by thestonefox** [comment]
One last thing, and this is totally optional.

But would you consider doing a bit of extra work on adding a unity event helper as in this PR:

https://github.com/thestonefox/SteamVR_Unity_Toolkit/pull/509

Basically, it's a script that you attach to the same object as your delegate event script and it allows you to use unity events.

It's something I'm trying to support as standard.

If it's not making sense or you don't really have time then no worries I'll merge this and add the Unity Event helper afterwards :)

**Comment by mbbmbbmm** [comment]
I looked at it, but unfortunately I am quite new to events in general and don't really understand how to deal with the event args. I'd love to learn it but at the moment the problem is the time. But I promise to examine how you add it so I'll be able do it myself in the future!

**Comment by thestonefox** [comment]
ok no worries, then lets get this merged in!

**Comment by mbbmbbmm** [comment]
Very cool! Thank you!

**Pull request merged**